{Release} Upgrade to Azure CLI 2.76.0 (#31884)
Azure/azure-cli: src/azure-cli/HISTORY.rst

* `az aks create/update`: Add `--node-provisioning-mode` and `--node-provisioning-default-pools` parameters (#31836)
* `az aks safeguards`: Add command group to manage deployment safeguards (#31793)
* `az aks create`: Support the Azure Monitor metrics and logs addon functionality (#31828)
* `az aks create`: Add new parameter `--disable-run-command` to disable run command feature for the cluster (#31854)
* `az aks update`: Add new parameters `--disable-run-command` and `--enable-run-command` to toggle the run command feature on or off (#31854)
* `az aks update`: Support updating load balancer sku from basic to standard (#31874)

**App Config**

* `az appconfig create/update`: Add support for setting key value revision retention period (#31725)
* `az appconfig feature list/delete/set`: Support filtering by tags (#31711)

**App Service**

* `az webapp create`: Add `--domain-name-scope` parameter to support DNL scopes during site creation (#31863)
* `az webapp sitecontainers convert`: Add a command to switch between sitecontainers and classic (#31790)

**ARM**

* `az group export`: Add `--export-format` to specify the format of the exported template (#31667)

> * Fix #31709: `az deployment`: Fix `the content for this response was already consumed` when create resource (#31722)

Reviewer: ```suggestion
* Fix #31709: `az deployment`: Fix `the content for this response was already consumed` error when create resource (#31722)
```

* `az deployment create/validate/what-if`: Expose `--validation-level` parameter at all scopes (#31747)

**Backup**

* `az backup restore restore-disks`: Add new parameter `--cvm-os-des-id` to specify the Disk Encryption Set ID to use for OS disk encryption (#31853)

**Cloud**

* Change `active_directory_graph_resource_id` endpoint of `AZURE_US_GOV_CLOUD` to `https://graph.microsoftazure.us/` (#31849)

**Compute**

* `az sig in-vm-access-control-profile`: Add command group to manage in VM access control profile (#31720)
* `az sig in-vm-access-control-profile-version`: Add command group to manage in VM access control profile version (#31720)
* `az vm disk attach`: Add new parameters `--source-snapshots-or-disks` `--source-disk-restore-point` to support implicit disk creation from snapshot and disk restore points (#31830)
* `az vmss create`: Add new parameter `--enable-automatic-repairs` to support setting automatic repairs policy (#31722)

> * Fix #31685: `az vm/vmss update` Fix error with `NoneType` object has no attribute `mode` when update wireserver profile (#31855)

Reviewer 2: ```suggestion
* Fix #31685: `az vm/vmss update`： Fix error with `NoneType` object has no attribute `mode` when update wireserver profile (#31855)
```

Release History
===============

2.76.0
++++++

**ACR**

* Fix #31572: `az acr check-health`: Fix crashes when the registry is not found (#31753)
* `az acr create`: Fix domain name label suffix validation (#31783)

**AKS**

* `az aks machine show/list`: Add zones to table output (#31668)
* `az aks`: Support Virtual Machines node pools in AKS commands (#31780)
* `az aks update`: Support VMAS to VMS agent pool migration in AKS commands (#31831)
* `az aks create/update`: Add new parameter `--enable-ai-toolchain-operator` to enable Kaito addon (#31485)
* `az aks`: Add `--node-provisioning-mode` and `--node-provisioning-default-pools` parameters (#31836)
* `az aks safeguards`: Add command group to manage deployment safeguards (#31793)

> * `az aks create`: Update prom and logs addon to use sdk (#31828)

Reviewer: ```suggestion
* `az aks create`: Support the Azure Monitor metrics and logs addon functionality (#31828)
```

* `az aks create`: Add new parameter `--disable-run-command` to disable run command feature for the cluster (#31854)

> * `az aks update`: Add  new parameters `--disable-run-command` and `--enable-run-command` to toggle the run command feature on or off (#31854)

Reviewer 3: ```suggestion
* `az aks update`: Add new parameters `--disable-run-command` and `--enable-run-command` to toggle the run command feature on or off (#31854)
```

* `az aks update`: Support updating load balancer sku from basic to standard (#31874)

**App Config**

* `az appconfig create/update`: Add support for setting key value revision retention period (#31725)
* `az appconfig feature list/delete/set`: Support filtering by tags (#31711)

**App Service**

* `az webapp create`: Add `--domain-name-scope` parameter to support DNL scopes during site creation (#31863)
* `az webapp sitecontainers convert`: Add a command to switch between sitecontainers and classic (#31790)

**ARM**

* `az group export`: Add `--export-format` to specify the format of the exported template (#31667)
* Fix #31709: `az deployment`: Fix `the content for this response was already consumed` when create resource (#31722)

> * `az deployment`: Expose `--validation-level` parameter at all scopes for `create`, `validate`, and `what-if` subcommands (#31747)

Reviewer 4: ```suggestion
* `az deployment create/validate/what-if`: Expose `--validation-level` parameter at all scopes (#31747)
```

**Cloud**

* Change `active_directory_graph_resource_id` endpoint of `AZURE_US_GOV_CLOUD` to `https://graph.microsoftazure.us/` (#31849)

**Compute**

* `az sig in-vm-access-control-profile`: Add command group to manage in VM access control profile (#31720)
* `az sig in-vm-access-control-profile-version`: Add command group to manage in VM access control profile version (#31720)
* `az vm disk attach`: Add new parameters `--source-snapshots-or-disks` `--source-disk-restore-point` to support implicit disk creation from snapshot and disk restore points (#31830)
* `az vmss create`: Add new parameter `--enable-automatic-repairs` to support setting automatic repairs policy (#31722)
* Fix issue #31685: `az vm/vmss update` Fix error with `NoneType` object has no attribute `mode` when update wireserver profile (#31855)
* `az vm disk attach`: Add support for setting disk size and sku of implicit disk creation from snapshot and disk restore points (#31862)
* `az compute-recommender spot-placement-score`: Add new command to replace original command `az compute-recommender spot-placement-recommender` (#31858)

**Consumption**

> 

Reviewer 2: ```suggestion
```

* `az consumption budget create`: Add command example (#31553)

> 

Reviewer 2: ```suggestion
```

**Network**

* `az network application-gateway waf-policy`: Support `computedDisabledRules` read-only property (#31755)
* `az network application-gateway waf-policy custom-rule create`: Support `GeoLocationXffHeader`, `ClientAddrXffHeader` for `groupByVariables` (#31767)
* Fix #31727: `az network private-link-service create`: Support multiple IP configurations (#31735)

**Packaging**

* Support RHEL 10 and CentOS Stream 10 (#31602)

**Profile**

* `az login`: Add `--claims-challenge` argument to support interactive authentication with claims challenge (#31778)

**RDBMS**

* `az postgres flexible-server replica create/promote`: Enable replica operations for elastic cluster operations (#31705)
* `az postgresql flexible-server create`: Handle failed IP address check (#31834)

**Service Fabric**

* `az sf managed-cluster network-security-rule`: Add new parameter `--source-addr-prefix` to specify the CIDR or source IP range (#31714)
* `az sf managed-cluster network-security-rule`: Add new parameter `--dest-addr-prefix` to specify the destination port or range (#31714)

> * `az sf managed-cluster network-security-rule`: Add  new parameter `--source-port-range` to specify the CIDR or source IP range (#31714)

Reviewer 2: ```suggestion
* `az sf managed-cluster network-security-rule`: Add new parameter `--source-port-range` to specify the CIDR or source IP range (#31714)
```

> * `az sf managed-cluster network-security-rule`: Add  new parameter `--dest-port-range` to specify the destination address prefix (#31714)

Reviewer 2: ```suggestion
* `az sf managed-cluster network-security-rule`: Add new parameter `--dest-port-range` to specify the destination address prefix (#31714)
```
